### PR TITLE
Experimental i2c0 bus support

### PR DESCRIPTION
**Pull request by RogueM**

### CODE REVIEW

**Comment by RogueM** [comment]
the idea is that it should be possible to:

1) support enviro-phat when it is hooked up to the i2c0 bus, for example to avoid address conflict on the main bus (say combining Explorer pHAT with rev1 of Enviro pHAT)

2) support two units, one on i2c1 and one on i2c0, say to sample different areas (or as per botlab reference QA readings)

goal 1 is achieved by creating an 'altbus' object if it's possible to create an instance of the SMBus class on the i2c0 bus. If it is, then the 'regular' instances of class for all the enviro-phat sensors will control the chips detected on the i2c0.

goal 2 is an extension of the above: if instances of the 'regular' classes can be made on the primary i2c1 bus, then 'alt'-classes are created to control the chips detected on the i2c0. Those are NOT created if either 'altbus' or the (all) 'alt'-classes can't be created.

... note that the 'alt'-classes can be addressed whether there is an enviro-phat can be detected on the primary i2c1 bus, so that code that uses those can be run without modification. They WON'T however work if the only Enviro pHAT detected is on the primary bus, to avoid confusion.

In other words, the 'regular' classes can be used to control Enviro pHAT on either busses (as long as i2c0 is set up correctly), and translate seamlessly when the pHAT is moved from one bus to the other. The 'alt' classes however are reserved to 'altbus' and will not translate to the primary bus.

**Comment by RogueM** [comment]
Also, the leds class is shadowed by an 'alt'class itself, addressed by BMC5 (simply because it is physically close to i2c0, and is unlikely to mess any other pHAT or HAT - well, OK, Explorer HAT uses it for LED4, but that's an OUTPUT too, so not a huge issue).

Anyhow, I'm not sure it all make sense, but that's the broad idea of the proof of concept, even though having two units attached to the same Pi is not a typical use case. It demonstrate a principle that could be applied to other things like the Micro Dot pHAT or Scroll pHAT, just to name two use cases I'd expect to be popular if full i2c0 support was available there.

**Comment by Gadgetoid** [comment]
In the case where the second i2c bus is disabled, but the user attempts to import "altx" what happens? Will this raise an import error? Since those objects wont exist. It might be worth defining them as `None` by default to allow code patterns like:

``` python
from envirophat import altweather

if altweather is not None:
    # Do something
```

**Comment by RogueM** [comment]
yes, probably. I know I added some logic in the scrollphat examples I did at the weekend, but yes, I agree, this should be handled at the library level, as gracefully as possible.
